internal/provider/kubernetes/controller.go

```diff
@@ -1317,7 +1322,10 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 	}
 
 	// Watch Gateway CRUDs and reconcile affected GatewayClass.
-	gPredicates := []predicate.Predicate{predicate.NewPredicateFuncs(r.validateGatewayForReconcile)}
+	gPredicates := []predicate.Predicate{
+		predicate.NewPredicateFuncs(r.validateGatewayForReconcile),
+		predicate.GenerationChangedPredicate{},
```

Comment:
this seems like a very useful thing to do, should we add this predicate to all resources as a separate PR ?
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate#GenerationChangedPredicate

Reply:
Sure, I will create a separate PR for it, during tests I have found many reconciliations in the logs on Gateway status updates.

Reply:
xref: https://github.com/envoyproxy/gateway/pull/1988